### PR TITLE
[n/a] feat: endpoint to retrieve all authors 

### PR DESCRIPTION
**Pull request by IanSeng**
# Summary 
This pr introduced a new endpoint to be used for retrieving all authors and peer-to-peer server connection.

**Documentation**:  [followers wikipage](https://github.com/C404TEAMW21/CMPUT404_PROJECT/wiki/Followers)

## Sub-tasks 
- [x] All author endpoint (excluding superuser and not admin approved author)
- [x] Refrain the users from using the endpoint if their account is not admin approved
- [x] Unit testing
- [x] Cleaned up old code 
- [x] Update documentation 

## Resolve
Resolve n/a

### CODE REVIEW

**Review by HailanXyouknow — DISMISSED** [review]
LGTM! 😃

File: server/author/tests/test_create_author_endpoint.py [hunk 1]
```diff
@@ -144,6 +147,7 @@ class TestAuthGetAuthorEndpoint(TestCase):
     """Test API(GET)://service/author/{AUTHOR_ID}/"""
     def setUp(self):
         self.client = APIClient()
+        print(AUTH_USER_URL)
```
Comment (keanweng97) [inline server/author/tests/test_create_author_endpoint.py:150]: Is this line for debugging?

File: server/author/views.py [hunk 2]
```diff
@@ -58,3 +58,16 @@ def get_object(self):
...
+    serializer_class = AuthorProfileSerializer
+    authenticate_classes = (authentication.TokenAuthentication,)
+    permission_classes = (permissions.IsAuthenticated,)
+    
+    def get_queryset(self):
+        if not self.request.user.adminApproval:
+            raise AuthenticationFailed(
+                detail={"error": ["User has not been approved by admin"]})
+        
+        return get_user_model().objects.filter(type='author')
```
Comment (keanweng97) [inline server/author/views.py:73]: Does this get all authors? Including unapproved authors?
Maybe we should check for approved authors

**Review by keanweng97 — COMMENTED** [review]
Other than that, LGTM! Thanks Ian!

File: server/author/tests/test_create_author_endpoint.py (hunk 1 above)
Comment (IanSeng) [reply server/author/tests/test_create_author_endpoint.py]: Good catch!!!

File: server/author/views.py (hunk 2 above)
Comment (IanSeng) [reply server/author/views.py]: this exclude superuser, hmmm im not sure if we want to filter out user without admin approval but I cool with either way.

**Review by keanweng97 — APPROVED** [review]
LGTM 🌟

**Pull request merged**